transfers/waterlevels_transducer_transfer.py

```diff
@@ -13,149 +13,196 @@
+        # convert nodeployments to errors
+        for pointid, (min_date, max_date) in nodeployments.items():
+            self._capture_error(
+                pointid,
+                "DateMeasured",
+                f"no deployment between {min_date} and {max_date}",
             )
-            try:
-                session.commit()
-            except Exception as e:
-                errors.append({"pointid": pointid, "error": e})
-                logger.critical(
-                    f"Error committing water levels {release_status} block: {e}"
-                )
-                session.rollback()
-                continue
 
-    return input_df, cleaned_df, errors
+    def _make_observation(
+        self,
+        pointid: str,
+        row: pd.Series,
+        release_status: str,
+        deps_sorted: list,
+        nodeployments: dict,
+    ) -> TransducerObservation | None:
+        deployment = _find_deployment(row.DateMeasured, deps_sorted)
+
+        if deployment is None:
+            if pointid not in nodeployments:
+                nodeployments[pointid] = (row.DateMeasured, row.DateMeasured)
+            else:
+                min_date, max_date = nodeployments[pointid]
+                if row.DateMeasured < min_date:
+                    min_date = row.DateMeasured
+                elif row.DateMeasured > max_date:
+                    max_date = row.DateMeasured
+                nodeployments[pointid] = min_date, max_date
+
+            logger.critical(
+                f"No deployment found for PointID={pointid} at {row.DateMeasured}"
+            )
+            return None
+
+        try:
+            payload = dict(
+                parameter_id=self.groundwater_parameter_id,
+                deployment_id=deployment.id,
+                observation_datetime=row.DateMeasured,
+                value=row.DepthToWaterBGS,
+                release_status=release_status,
+            )
+            obspayload = CreateTransducerObservation.model_validate(
+                payload
+            ).model_dump()
+            return TransducerObservation(**obspayload)
+
+        except ValidationError as e:
+            logger.critical(f"Observation validation error: {e.errors()}")
+            self._capture_error(pointid, str(e), "DepthToWaterBGS")
+
+
+class WaterLevelsContinuousPressureTransferer(WaterLevelsContinuousTransferer):
+    source_table = "WaterLevelsContinuous_Pressure"
+    _partition_field = "QCed"
+    _sensor_type = "Pressure Transducer"
+
+
+class WaterLevelsContinuousAcousticTransferer(WaterLevelsContinuousTransferer):
+    source_table = "WaterLevelsContinuous_Acoustic"
+    _partition_field = "PublicRelease"
+    _sensor_type = "Acoustic Sounder"
+
+
+def _find_deployment(ts, deployments):
+    for d in deployments:
+        start = Timestamp(d.installation_date)
+        if start > ts:
+            break  # because sorted by start
+        end = Timestamp(d.removal_date) if d.removal_date else Timestamp.max
```

Comment:
When `removal_date` gets converted to `pandas.Timestamp` it sets the time to midnight (e.g. `2018-01-01` --> `2018-01-01T00:00:00`). During comparison if there are measurements on that date, like `2018-01-01T12:00:00`, it will evaluate that the measurement occurs after removal and therefore won't be transferred. Is this the expected behavior? Or, should we compare the timestamp of the measurement to the day after removal?